crates/router/src/connector/worldpay.rs

```diff
@@ -86,10 +92,14 @@ impl ConnectorCommon for Worldpay {
         res: Response,
         event_builder: Option<&mut ConnectorEvent>,
     ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
-        let response: WorldpayErrorResponse = res
-            .response
-            .parse_struct("WorldpayErrorResponse")
-            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        router_env::logger::info!(raw_response=?res);
+        let response = if res.status_code != 404 {
+            res.response
+                .parse_struct("WorldpayErrorResponse")
+                .change_context(errors::ConnectorError::ResponseDeserializationFailed)?
+        } else {
+            WorldpayErrorResponse::default()
```

Comment:
Can we not default in status mappings?

Reply:
They send an empty body for 404 status codes .. we need to conditionally parse due to this behaviour or else it would throw derserialization errors leading to 5xx on our end

Making every field optional in error response also wouldn't work since it would expect `{}` but WorldPay empty body ``